masq/src/terminal_interface.rs

```diff
@@ -93,18 +87,43 @@ impl Clone for TerminalWrapper {
 fn result_wrapper_for_in_memory_terminal() -> std::io::Result<MemoryTerminal> {
     Ok(MemoryTerminal::new())
 }
-
-fn configure_interface<F, U, E: ?Sized, D>(
-    interface_raw: Box<F>,
-    terminal_type: Box<E>,
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//This construction, including those functions with closures above that are strongly affected by the way how interface_configurator()
+//is written, is so complicated because there are tough obstacles to write simple tests here.
+//interface_configurator() substitutes something that may otherwise look, in the production code, simplified, like:
+//let terminal = match DefaultTerminal::new() {*something*};
+//let interface = match Interface::start_with("masq",terminal){*something*};
+//if let Err(e) = interface.set_prompt(){*something*};
+//Ok(TerminalReal::new(interface))
+//
+//However, since we want to write tests we have to face here the following:
+//1) DefaultTerminal alone is a part which can be theoretically tested outside, with certain troubles, but because of the other
+//lines of codes in interface_configurator() it makes sense for it to stay there.
+//2) It's quite hard to simulate a failure at Interface::start_with, though possible; you have to implement your own "terminal
+//type", which you'll then instruct to cause an error during a call of the function below. It requires an implementation
+//of linefeed::Terminal and creation of some other objects, hanged on it and dependant on each other, which leads up to an extremely
```

Comment:
Perhaps "hung" would be better. "Hung" means "suspended;" "hanged" means "suspended by the neck until dead." Also, it's "dependent" instead of "dependant."